internal/controller/scheduler/pod_watcher.go

```diff
@@ -373,21 +372,29 @@ func (w *podWatcher) formatImagePullFailureMessage(statuses []corev1.ContainerSt
 	return "The following images could not be pulled or were unavailable:\n\n" + tw.Render()
 }
 
-func (w *podWatcher) evictPod(ctx context.Context, log *zap.Logger, pod *corev1.Pod, jobUUID uuid.UUID) {
-	eviction := &policyv1.Eviction{
-		ObjectMeta: pod.ObjectMeta,
+func (w *podWatcher) forcefullyDeletePod(ctx context.Context, log *zap.Logger, podMetadata *metav1.ObjectMeta, reason string) error {
+	// Force immediate deletion with zero grace period
+	// We generally believe when our controller decide a payload should go, there shouldn't be good reason to keep it waiting for
+	// graceful timeout.
+	//
+	// Some organizations conduct massive job cancelling, without this, it will cause unnecessary cluster resource saturation.
+	deleteOptions := metav1.DeleteOptions{
+		GracePeriodSeconds: new(int64), // zero value
 	}
-	if err := w.k8s.PolicyV1().Evictions(w.cfg.Namespace).Evict(ctx, eviction); err != nil {
-		podEvictionErrorsCounter.WithLabelValues("image_pull_failure", string(kerrors.ReasonForError(err))).Inc()
-		log.Error("Couldn't evict pod", zap.Error(err))
-		return
+
+	if err := w.k8s.CoreV1().Pods(podMetadata.Namespace).Delete(ctx, podMetadata.Name, deleteOptions); err != nil {
+		log.Error("Couldn't forcefully delete pod", zap.Error(err))
+		forcefulPodDeletionErrorsCounter.WithLabelValues(reason, string(kerrors.ReasonForError(err))).Inc()
+		return err
+	}
+	forcefullyDeletedPodCounter.WithLabelValues(reason).Inc()
+
+	// Mark the job as ignored to avoid further processing
+	if jobUUID, err := jobUUIDForObject(podMetadata); err == nil {
+		w.ignoreJob(jobUUID)
```

Comment:
`if err == nil` is very unusual Go - prefer to indent the error flow instead

```suggestion
	// Mark the job as ignored to avoid further processing
	jobUUID, err := jobUUIDForObject(podMetadata)
	if err != nil {
		return nil
	}
	w.ignoreJob(jobUUID)
```